backend/server.js

```diff
@@ -67,6 +73,36 @@ app.get("/data/*", async (req, resp) => {
   }
 });
 
+// See https://cloud.google.com/iap/docs/signed-headers-howto
+async function checkIap(req, resp, next) {
+  let iapJwt = req.header("x-goog-iap-jwt-assertion");
+  if (!iapJwt) {
+    resp.status(401).send("Missing x-goog-iap-jwt-assertion header");
+    return;
+  }
+
+  try {
+    // TODO Can we cache this between requests?
+    let iapPublicKeys = await oauthClient.getIapPublicKeys();
```

Comment:
I think these are Google public keys. I wouldn't expect them to change much, but I haven't found any docs/guidance yet about caching.